### DISCUSSION TITLE
Artifact generated on Joint TTS

### DISCUSSION THREAD

**Discussion by Fhrozen** [espnet/espnet#5070] [Q&A] [answered]
@kan-bayashi 
I am wondering if you have any issues with joint TTS.
I've been trying to train a model with Prodiff on Tsukuiyomi. 
First, I trained the TTS model on JVS. But the adaptation and the vocoder (hifigan) produced artifacts, and the quality was not good enough. So, I decided to move to full band (44100Hz + Fmin 0, nfft 2048, shift 300), but again, artifacts were present (despite training a new Hifigan for full band).
Finally, I tried to train a joint model using shift 512,  but it does not work well despite using a different vocoder during fine-tuning. 
These are the config files for the joint training
[finetune_joint_prodiff_stylegan.yaml.txt](https://github.com/espnet/espnet/files/11117847/finetune_joint_prodiff_stylegan.yaml.txt)
[finetune_joint_prodiff_hifigan.yaml.txt](https://github.com/espnet/espnet/files/11117848/finetune_joint_prodiff_hifigan.yaml.txt)
This is a sample on JSUT after 222 epochs. May seem few, but I pretrained the TTS model, and the adv and disc loss are almost 0
[0222ep.zip](https://github.com/espnet/espnet/files/11117869/0222ep.zip)

Figs:

![text2mel_loss](https://user-images.githubusercontent.com/11988996/229015318-29f5ca4d-71e3-44ac-9cd6-b0a72a7eed5d.png)
![adv_loss](https://user-images.githubusercontent.com/11988996/229015323-5a991814-b377-4190-8a53-390968bcbf1b.png)
![discriminator_loss](https://user-images.githubusercontent.com/11988996/229015325-3922a3b0-327a-4125-8c01-d8ebac889ee6.png)
![duration_loss](https://user-images.githubusercontent.com/11988996/229015327-8c16e9c8-aaa0-43ae-98c8-a2af1ae7cd46.png)
![energy_loss](https://user-images.githubusercontent.com/11988996/229015328-e6351df3-443e-4ebd-953a-9221a8471ced.png)
![fake_loss](https://user-images.githubusercontent.com/11988996/229015333-cd747454-8fdb-43ae-bf08-5b21f5ee96db.png)
![feat_match_loss](https://user-images.githubusercontent.com/11988996/229015335-1272feac-8c3f-4243-868f-e3dd91eba72e.png)
![l1_loss](https://user-images.githubusercontent.com/11988996/229015340-47818163-892b-4c6d-9951-bd39ef740b93.png)
![loss](https://user-images.githubusercontent.com/11988996/229015343-d7e2015a-fbba-43a7-942a-f1ea5c79e66e.png)
![mel_loss](https://user-images.githubusercontent.com/11988996/229015344-a718e432-c8b7-4a8f-85aa-6b419ee7eb61.png)
![pitch_loss](https://user-images.githubusercontent.com/11988996/229015345-672d6f0a-e1cb-4fc0-b5d9-bf4c1d2e84a4.png)
![real_loss](https://user-images.githubusercontent.com/11988996/229015347-2aafe625-4e44-4c34-956e-e39d94193f1e.png)
![ssim_loss](https://user-images.githubusercontent.com/11988996/229015350-6cae0b41-577b-4d3e-9e49-80ae9c8db7da.png)

Any Idea what I am doing wrong?

**Comment by kan-bayashi** [comment, marked as answer]
The easiest solution is finetuning the vocoder using teacher forced TTS outputs.
This can almost reduce the metallic artifacts.
You can check the comparison results in Table1.
https://arxiv.org/pdf/2110.07840.pdf

From the training curve, it seems the discriminator got much stronger than the generator.
You need to tune the learning rate of discriminator.
